RHDEVDOCS-3423-managing-user-defined-alerting-rules (#48754)
openshift/openshift-docs: modules/monitoring-managing-core-platform-alerting-rules.adoc

//
// * monitoring/managing-alerts.adoc

:_content-type: CONCEPT
[id="managing-core-platform-alerting-rules_{context}"]
= Managing alerting rules for core platform monitoring

:FeatureName: Creating and modifying alerting rules for core platform monitoring
include::snippets/technology-preview.adoc[leveloffset=+1]

{product-title} {product-version} monitoring ships with a large set of default alerting rules for platform metrics.
As a cluster administrator, you can customize this set of rules in two ways:

* Modify the settings for existing platform alerting rules by adjusting thresholds or by adding and modifying labels.
For example, to help you route and triage issues flagged by an alert, you can change the `severity` label for an alert from `warning` to `critical`.

* Define and add new custom alerting rules by constructing a query expression based on core platform metrics in the `openshift-monitoring` namespace.

.Core platform alerting rule considerations

* New alerting rules must be based on the default {product-title} monitoring metrics.

* You can only add and modify alerting rules. You cannot create new recording rules or modify existing recording rules.

> * If you modify existing platform alerting rules, your modifications are not reflected in the Prometheus alerts API. 
> Therefore, currently dropped alerts still appear in the {product-title} web console even though they were forwarded to Alertmanager. 
> Additionally, any modifications made to alerts such as a changed severity label do not appear in the web console.

Reviewer: ```suggestion
* If you modify existing platform alerting rules though an `AlertRelabelConfig` object, your modifications are not reflected in the Prometheus alerts API. 
Therefore, any dropped alerts still appear in the {product-title} web console even though they are no longer forwarded to Alertmanager. 
Additionally, any modifications made to alerts such as a changed severity label do not appear in the web console.
```
Sorry about the late correction on this. I had to have a closer look myself to confirm how this works. Based on https://github.com/openshift/enhancements/pull/1030/files#r811793390 and https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alert_relabel_configs.

Author: Changed to read:

```
* If you modify existing platform alerting rules by using an `AlertRelabelConfig` object, your modifications are not reflected in the Prometheus alerts API. 
Therefore, any dropped alerts still appear in the {product-title} web console even though they are no longer forwarded to Alertmanager. 
```

> Additionally, any modifications made to alerts such as a changed severity label do not appear in the web console.

Reviewer 2: ```suggestion
Additionally, any modifications to alerts, such as a changed `severity` label, do not appear in the web console.
```

Author: fixed

// Module included in the following assemblies:
//
// * monitoring/managing-alerts.adoc

:_content-type: CONCEPT
[id="managing-core-platform-alerting-rules_{context}"]
= Managing alerting rules for core platform monitoring

> :FeatureName: User-defined and user-modified alerting rules for core platform monitoring

Reviewer: Not sure if we can drop the `user-defined` here without creating more confusion. Happy to leave this as is, but maybe someone has a good idea.

Author: I changed it to `Adding and modifying alerting rules for core platform monitoring`

include::snippets/technology-preview.adoc[leveloffset=+1]

{product-title} {product-version} monitoring ships with a large set of default alerting rules for platform metrics.
As a cluster administrator, you can customize this set of rules in two ways:

> * Modify the settings for existing platform alerting rules by adjusting thresholds or by adding and modifying labels.

Reviewer 3: note: only AlertingRule resources created in the `openshift-monitoring` namespace will be reconciled by CMO

For example, to help you route and triage issues flagged by an alert, you can change the `severity` label for an alert from `warning` to `critical`.

* Define and add new custom alerting rules by constructing a query expression based on core platform metrics in the `openshift-monitoring` namespace.

> After this new, user-defined rule exists, you can view, edit, and remove it.

Reviewer 3: `After this new, user-defined rule exists`
I am kindly think the expression is not good, but don't know what's the better expression

Author: I've deleted this sentence. It doesn't add any useful information that a user doesn't already know,

.Core platform alerting rule considerations

> * User-defined alerting rules must be based on the default {product-title} monitoring metrics.

Reviewer: ```suggestion
* New custom alerting rules must be based on the default {product-title} monitoring metrics.
```

Author: Changed per your suggestion

> * You can only modify alerting rules. You cannot create new recording rules or modify existing recording rules.

Reviewer: ```suggestion
* You can only add and modify alerting rules. You cannot create new recording rules or modify existing recording rules.
```

Author: Changed per your suggestion

* If you modify existing platform alerting rules, your modifications are not reflected in the Prometheus alerts API. 

> Therefore, currently dropped alerts still appear in the {product-title} web console even though they were not forwarded to Alertmanager. 

Reviewer 3: even though they were not forwarded to Alertmanager
->
even though they were forwarded to Alertmanager
